src/lib/dnssd/Resolver.h

```diff
@@ -66,6 +66,19 @@ struct ResolvedNodeData
     Optional<uint32_t> GetMrpRetryIntervalIdle() const { return mMrpRetryIntervalIdle; }
     Optional<uint32_t> GetMrpRetryIntervalActive() const { return mMrpRetryIntervalActive; }
 
+    bool IsDeviceSleepy() const
+    {
+        // If either retry interval (Idle - CRI, Active - CRA) has a value and that value
+        // is greater than the default, then it is a Sleepy End Device (SED)
```

Comment:
I think we should not state that device is SED in such direct way. Basically we don't know that, we just assume that it will be treated as a SED from a communication point of view, because we may not manage to develop something better before 1.0. Could you consider stating in more clear way it is just an assumption? Like `IsDeviceTreatedAsSleepy` or something like that?

I think it can protect from using this method to determine SED device somewhere we should not base only on that assumption, without complete knowledge if device is battery operated and etc.

Reply:
@kkasperczyk-no, I think that is a fair statement. A non-sleepy device whose values cause this function to return true would still behave the correct way (i.e. quick responses to reads/writes/etc.). The other devices on the Fabric would invoke their 'SED logic' for that device, but that logic would not end up being needed.

@pan-apple , @billwatersiii, Skip, do you agree or have concerns?

Reply:
That seems fine for 1.0.  If a device advertises a slow polling rate we need to treat it as a sleepy if it is or not.

Reply:
Yes, this sounds good to me.

Reply:
I pushed the update